### PR TITLE
Switch to TypeScript

### PR DESCRIPTION
**Pull request by Eric-Arellano**
## Motivation

See https://github.com/ParkingReformNetwork/organization-dashboard/pull/33. 

I at first held off on adding TypeScript to keep things simple, since there was so little JavaScript code. But we now have a non-trivial amount of code, including some complex things like our filter logic and taking functions as arguments.

TypeScript makes the code easier to read becauses it's more self-documenting and IDEs are way more useful with auto-complete. It also catches a whole class of bugs.

While there is a small learning curve with TypeScript, we found in the organization-dashboard that it was worth it to make the code more accessible for contributors, including beginners.

## How TypeScript interacts with our tools

Parcel ignores types and simply strips them. Playwright (test runner) also ignores them.

So we have a manual `npm run check` command that checks for any type issues. We run this in CI.

## Follow up PRs

This only adds the bare minimum config and changes any files to TypeScript that don't require making changes. That's to keep this PR easy to review and lower risk that we accidentally break something.

Followup PR(s) will add types to more files and eventually enable strict mode.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "start": "parcel",
     "build": "rm -rf dist; parcel build --detailed-report",
     "test": "playwright test",
+    "check": "tsc --noEmit",
```
Comment (Eric-Arellano) [inline package.json:10]: `tsc` is the official TypeScript compiler

File: src/js/fontAwesome.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {
 } from "@fortawesome/free-solid-svg-icons";
 import "@fortawesome/fontawesome-svg-core/styles.css";
 
-const setUpIcons = () => {
+const setUpIcons = (): void => {
```
Comment (Eric-Arellano) [inline src/js/fontAwesome.ts:13]: This is how you say in TypeScript that the function doesn't return anything

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (Eric-Arellano) [inline tsconfig.json:1]: We'll add options later like enabling strict mode. For now, we only need the file to exist

**Review by Alankvuong — APPROVED** [review]
Everything looks good to me!

**Pull request merged**